transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -60,17 +64,24 @@
                     DefaultChannelPipeline.class, MessageSizeEstimator.Handle.class, "estimatorHandle");
     final AbstractChannelHandlerContext head;
     final AbstractChannelHandlerContext tail;
+    final OrderedEventExecutor executor;
 
     private final Channel channel;
     private final ChannelFuture succeededFuture;
     private final VoidChannelPromise voidPromise;
     private final boolean touch = ResourceLeakDetector.isEnabled();
+    private final List<AbstractChannelHandlerContext> handlers = new ArrayList<>(4);
```

Comment:
I decided to just use an `ArrayList` here and synchronize on it. The reason for it is that it greatly simplifies things. I don't expect a ChannelPipeline to be too large so scanning the while List at worst shouldn't be a big deal. We did the same before when walking the while linked-list.